src/main/java/picard/illumina/UnsortedBasecallsConverter.java

```diff
@@ -75,77 +77,21 @@ protected UnsortedBasecallsConverter(
-                tileReadExecutor.submit(new TileReadProcessor(tile));
-                tilesProcessing++;
-                tilesSubmitted++;
-            } else {
-                try {
-                    Thread.sleep(100);
-                } catch (InterruptedException e) {
-                    throw new PicardException("Tile processing thread interrupted: " + e.getMessage());
-                }
-            }
-        }
-        awaitTileProcessingCompletion();
-    }
-
-    /**
-     * TileProcessor is a Runnable that process all records for a given tile. It uses the underlying
-     * IlluminaDataProvider to iterate over cluster data for a specific tile. Records are added to a
-     * cache as they are read.
-     * <p>
-     * After the tile processing is complete it notifies the CompletedWorkChecker that data is ready
-     * for writing.
-     */
-    private class TileReadProcessor implements Runnable {
-        private final int tileNum;
-
-        TileReadProcessor(final int tileNum) {
-            this.tileNum = tileNum;
-        }
-
-        @Override
-        public void run() {
-            Queue<ClusterData> queue = new ArrayDeque<>();
-            final BaseIlluminaDataProvider dataProvider = factory.makeDataProvider(tileNum);
+    public void processTilesAndWritePerSampleOutputs(final Set<String> barcodes) throws IOException {
+        for (Integer tile : tiles){
+            final BaseIlluminaDataProvider dataProvider = factory.makeDataProvider(tile);
 
             while (dataProvider.hasNext()) {
                 final ClusterData cluster = dataProvider.next();
-                readProgressLogger.record(null, 0);
                 if (includeNonPfReads || cluster.isPf()) {
-                    queue.add(cluster);
+                    barcodeRecordWriterMap.get(cluster.getMatchedBarcode()).write(converter.convertClusterToOutputRecord(cluster));
+                    progressLogger.record(null, 0);
                 }
             }
-
             dataProvider.close();
-            notifyWorkComplete(tileNum, Collections.singletonList(new RecordToWriterPump(queue)));
-        }
-    }
-
-    private class RecordToWriterPump implements Runnable {
-        private final Queue<ClusterData> clusterData;
-
-        RecordToWriterPump(final Queue<ClusterData> clusterData) {
-            this.clusterData = clusterData;
         }
-
-        @Override
-        public void run() {
-            ClusterData cluster;
-            while ((cluster = clusterData.poll()) != null) {
-                if (includeNonPfReads || cluster.isPf()) {
-                    barcodeRecordWriterMap.get(cluster.getMatchedBarcode()).write(converter.convertClusterToOutputRecord(cluster));
-                    writeProgressLogger.record(null, 0);
-                }
-            }
-            tilesProcessing--;
+        for(Writer<CLUSTER_OUTPUT_RECORD> writer : barcodeRecordWriterMap.values()){
+            writer.close();
```

Comment:
Same goes for here to prefer closing via the pool to closing each writer individually.